test-data/unit/check-classes.test

```diff
@@ -6891,3 +6891,68 @@ class B(A):  # E: Final class __main__.B has abstract attributes "foo"
 [case testUndefinedBaseclassInNestedClass]
 class C:
     class C1(XX): pass  # E: Name "XX" is not defined
+
+[case testExpandImportedMethodInClass]
+# https://github.com/python/mypy/issues/7045
+class Foo:
+    from a import meth
+
+reveal_type(Foo().meth)  # N: Revealed type is "def (x: builtins.int) -> builtins.int"
+[file a.py]
+def meth(self, x: int) -> int:
+    ...
+
+[case testExpandImportedFunctionInClass]
+class Foo:
+    from a import func
+
+reveal_type(Foo.func)  # N: Revealed type is "def () -> builtins.str"
+[file a.py]
+def func() -> str:
+    ...
+
+[case testExpandImportedMethodInNestedClass]
+class Foo:
+    class Bar:
+        from a import meth
+
+reveal_type(Foo.Bar().meth)  # N: Revealed type is "def (x: builtins.int) -> builtins.int"
+[file a.py]
+def meth(self, x: int) -> int:
+    ...
+
+[case testExpandUndefinedImports]
+class Foo:
+    from not_a_module import func
+
+reveal_type(Foo.func)
+reveal_type(Foo().func)
+[out]
+main:2: error: Cannot find implementation or library stub for module named "not_a_module"
+main:2: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
+main:4: note: Revealed type is "Any"
+main:5: note: Revealed type is "Any"
+
+[case testExpandImportedMethodWithFollowImports]
+# flags: --follow-imports=skip
+class Foo:
+    from a import meth
+
+reveal_type(Foo().meth)  # N: Revealed type is "Any"
+[file a.py]
+def meth(self, x: int) -> int:
+    ...
+
+[case testExpandWithNonSelfFirstArgument]
+class Foo:
+    from a import func
+
+reveal_type(Foo().func)
```

Comment:
Test what happens if we import two functions with the same name (from different modules)?

Also test what happens if we import a function over an existing name.

Test multiple classes that import the same function.

Test calling the original function via module and instance.